lib/rspec/resembles_json_matchers/resembles_any_of_matcher.rb

```diff
@@ -13,7 +13,7 @@ def initialize(expected)
     end
 
     def matches?(actual)
-      Array.wrap(actual).flatten.all? do |a|
+      matches_arity?(actual) && Array.wrap(actual).flatten.all? do |a|
```

Comment:
Rather than `arity`. which to me implies the number of args a method takes, perhaps just be explicit:

```ruby
actual = Array.wrap(actual)
if actual.empty?
  expected_matchers.empty?
else
  all? do |a|
    # ...
 end
end
```

Thoughts?